packages/openapi-fetch/test/index.with.dispatcher.test.ts

```diff
@@ -0,0 +1,1608 @@
+import { HttpResponse, type StrictResponse } from "msw";
+import createClient, { type Middleware, type MiddlewareRequest, type QuerySerializerOptions } from "../src/index.js";
+import type { paths } from "./fixtures/api.js";
+import { server, baseUrl, useMockRequestHandler, toAbsoluteURL } from "./fixtures/mock-server.js";
+import { Agent } from "undici";
+
+const dispatcher = new Agent({
+  connect: {
+    rejectUnauthorized: false,
+  },
+});
+
+beforeAll(() => {
+  server.listen({
+    onUnhandledRequest: (request) => {
+      throw new Error(`No request handler found for ${request.method} ${request.url}`);
+    },
+  });
+});
+
+afterEach(() => server.resetHandlers());
+
+afterAll(() => server.close());
+
+describe("TypeScript checks", () => {
```

Comment:
Appreciate you adding so many test cases to this PR, but we **don’t** want to just flat-out copy the entire test suite just for one change. I would suspect most of the tests don’t actually test this code path at all; we probably only need 1 or 2 new tests to start.

There may be some confusion with the v7-beta.test.ts file; that’s a separate case. It’s because when openapi-typescript 7.x gets a stable release, we’ll have to make a breaking release for openapi-fetch as well, and I’m just making sure that transition is smooth. But again, for a single option, we don’t need to copy all the tests

Reply:
Removed the extra test file and just added one test under options specifying a custom dispatcher - without it hitting a 'bad cert' website it only validates that specifying a custom dispatcher does not break general functionality.